code/modules/mob/living/clothing/clothing.dm

```diff
@@ -0,0 +1,636 @@
+	attacktext = "attacks"
+	sound/attack_sound = 'sound/weapons/punch1.ogg'
+	//sound/miss_sound = 'sound/weapons/punchmiss.ogg'
+	//friendly = "nuzzles" //If the mob does no damage with it's attack
+	environment_smash = 0 //Set to 1 to allow breaking of crates,lockers,racks,tables; 2 for walls; 3 for Rwalls
+
+	// Appearance vars
+	var/appearance_name = "blue schoolgirl uniform"
+
+	var/list/stored_appearances = list(
+		//initial(/obj/item/clothing/under/color/grey.name) = /obj/item/clothing/under/color/grey,
+		"blue schoolgirl uniform" = /obj/item/clothing/under/schoolgirl,
+		)
+
+	var/list/stored_apparence_names = list(
+		//initial(/obj/item/clothing/under/color/grey.name),
+		"blue schoolgirl uniform"
+		)
+
+	var/list/powers_with_host = list(
+		/datum/action/generic/clothes_drink_blood,
+		/datum/action/generic/living_clothing/detach,
+		/datum/action/generic/clothes_grow_armour,
+		/datum/action/generic/clothes_grow_blade,
+		)
+
+	var/list/current_dynamic_powers = list()
+
+	// Attachment vars
+	var/obj/item/clothing/under/living/linked_clothes = null
+
+/mob/living/simple_animal/clothing/get_blood_id()
+	return "blood"
+
+/mob/living/simple_animal/clothing/New()
+	// I guess we want this?
+	// create_reagents(1000)
+
+	var/datum/action/generic/set_living_clothing_appearance/CC = new(src)
+	CC.Grant(src)
+
+	set_appearance(appearance_name)
+
+	// Can hold 500 units of blood
+	create_reagents(500)
+	..()
+
+// Handle dir change due to clicking around on screen
+/mob/living/simple_animal/clothing/setDir(newDir)
+	handleDir(newDir)
+	return ..()
+
+// Handle dir change due to movement
+/mob/living/simple_animal/clothing/Moved(atom/OldLoc, newDir)
+	handleDir(newDir)
+	return ..()
+
+/mob/living/simple_animal/clothing/proc/handleDir(newDir)
+	// TODO: we should maybe use icon rotations instead...
+	// TODO: is this mob scalable??
+	//if(newDir != dir)
+	var/angle = dir2angle(newDir)
+	var/matrix/ntransform = matrix()
+	ntransform.Turn(angle)
+	transform = ntransform
+
+/mob/living/simple_animal/clothing/proc/set_appearance(appearance_name = "")
+	if(appearance_name == "")
+		appearance_name = input("Select your appearance!", "Living Clothing Appearance", null, null) in stored_apparence_names
+
+	src.appearance_name = appearance_name
+
+	var/appearance_type = stored_appearances[appearance_name]
+
+	var/obj/item/clothing/C = new appearance_type()
+
+	icon = C.icon
+	icon_living = C.icon_state
+	icon_state = C.icon_state
+
+	if(linked_clothes)
+		linked_clothes.icon = C.icon
+		linked_clothes.icon_state = C.icon_state
+		linked_clothes.item_state = C.item_state
+		linked_clothes.item_color = C.item_color
+
+		if(linked_clothes.loc && istype(linked_clothes.loc,/mob/living/carbon/human))
+			var/mob/living/carbon/human/H = linked_clothes.loc
+			H.update_inv_w_uniform()
+
+	sparkle()
+
+	qdel(C)
+
+/mob/living/simple_animal/clothing/UnarmedAttack(atom/A, proximity)
+	var/mob/living/M = A
+	if(M)
```

Comment:
I'm pretty sure you need if(istype(M)) because byond will happily do the cast on the line above even if it's invalid

Could be wrong though